pandas/tests/indexes/test_base.py

```diff
@@ -363,6 +364,19 @@ def test_constructor_dtypes_timedelta(self):
                         pd.TimedeltaIndex(list(values), dtype=dtype)]:
                 tm.assert_index_equal(res, idx)
 
+    def test_constructor_uint64(self):
```

Comment:
Reply to your comment about testing sub-class (and your other comment <a href="https://github.com/pandas-dev/pandas/pull/14937#pullrequestreview-15037467">above</a>): I did not have a sub-class at all, so I will add that.